components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/feature/CustomTabSessionTitleObserver.kt

```diff
@@ -4,33 +4,47 @@
 
 package mozilla.components.feature.customtabs.feature
 
-import mozilla.components.browser.session.Session
+import mozilla.components.browser.state.state.CustomTabSessionState
 import mozilla.components.concept.toolbar.Toolbar
 
 /**
  * Sets the title of the custom tab toolbar based on the session title and URL.
  */
 class CustomTabSessionTitleObserver(
```

Comment:
nit: since this is no longer an observer (yay!) and there's very minimal code, I would probably just inline this logic in the feature, but we can also keep it as a remnant of the past. :)

Reply:
Yeah, at first I wanted to refactor more here. But then I felt like not changing too much at once. :) This one has an interesting sub problem where I remember the values to know which one has changed. There may be the opportunity to make this use case nicer and then refactor this here too.